dawn/test/unit-test/dawn/Optimizer/Passes/TestComputeMaxExtent.cpp

```diff
@@ -27,153 +29,62 @@ using namespace dawn;
-    // Report diganostics
-    if(compiler_.getDiagnostics().hasDiags()) {
-      for(const auto& diag : compiler_.getDiagnostics().getQueue())
-        std::cerr << "Compilation Error " << diag->getMessage() << std::endl;
-      throw std::runtime_error("compilation failed");
-    }
-
-    DAWN_ASSERT_MSG(stencilInstantiationMap.count("compute_extent_test_stencil"),
-                    "compute_extent_test_stencil not found in sir");
-
-    return stencilInstantiationMap["compute_extent_test_stencil"];
-  }
-};
-
-TEST_F(ComputeMaxExtents, test_stencil_01) {
-  const std::shared_ptr<iir::StencilInstantiation>& instantiation =
-      loadTest("input/compute_extent_test_stencil_01.sir");
-  const auto& metadata = instantiation->getMetaData();
+TEST(ComputeMaxExtents, test_stencil_01) {
+  /*
+  vertical_region(k_start, k_end) { out = in[i - 1]; }
+  */
+  auto instantiation = IIRSerializer::deserialize("input/compute_extent_test_stencil_01.iir");
   const std::unique_ptr<iir::IIR>& IIR = instantiation->getIIR();
-  const auto& stencils = IIR->getChildren();
-  ASSERT_TRUE((stencils.size() == 1));
-  const std::unique_ptr<iir::Stencil>& stencil = stencils[0];
-
-  ASSERT_TRUE((stencil->getNumStages() == 2));
-  auto exts = stencil->getFields();
-  EXPECT_EQ(exts.size(), 3);
-  int u_id = metadata.getAccessIDFromName("u");
-  int out_id = metadata.getAccessIDFromName("out");
-  int lap_id = metadata.getAccessIDFromName("lap");
-
-  EXPECT_EQ(exts.at(u_id).field.getExtentsRB(),
-            (iir::Extents(dawn::ast::cartesian, -2, 2, -2, 2, 0, 0)));
-  EXPECT_EQ(exts.at(out_id).field.getExtentsRB(),
-            (iir::Extents(dawn::ast::cartesian, 0, 0, 0, 0, 0, 0)));
-  EXPECT_EQ(exts.at(lap_id).field.getExtentsRB(),
-            (iir::Extents(dawn::ast::cartesian, -1, 1, -1, 1, 0, 0)));
-}
-
-TEST_F(ComputeMaxExtents, test_stencil_02) {
-  const std::shared_ptr<iir::StencilInstantiation>& instantiation =
-      loadTest("input/compute_extent_test_stencil_02.sir");
   const auto& metadata = instantiation->getMetaData();
-  const std::unique_ptr<iir::IIR>& IIR = instantiation->getIIR();
   const auto& stencils = IIR->getChildren();
-  ASSERT_TRUE((stencils.size() == 1));
-  const std::unique_ptr<iir::Stencil>& stencil = stencils[0];
-
-  ASSERT_TRUE((stencil->getNumStages() == 3));
-  auto exts = stencil->getFields();
-  EXPECT_EQ(exts.size(), 6);
-  int u_id = metadata.getAccessIDFromName("u");
-  int out_id = metadata.getAccessIDFromName("out");
-  int coeff_id = metadata.getAccessIDFromName("coeff");
-
-  EXPECT_EQ(exts.at(u_id).field.getExtentsRB(),
-            (iir::Extents(dawn::ast::cartesian, -2, 2, -2, 2, 0, 0)));
-  EXPECT_EQ(exts.at(out_id).field.getExtentsRB(),
-            (iir::Extents(dawn::ast::cartesian, 0, 0, 0, 0, 0, 0)));
-  EXPECT_EQ(exts.at(coeff_id).field.getExtentsRB(),
-            (iir::Extents(dawn::ast::cartesian, 0, 0, 0, 0, 0, 0)));
+  auto exts = stencils[0]->getFields();
```

Comment:
kinda nit-picky, but I'd probably call this `fields` or something instead of `exts`

Reply:
that is what reviews are here for, thanks